modules/container/claws/container/vect.hpp

```diff
@@ -1,241 +1,364 @@
+#define CLAWS_VECT_OPERATOR_DEF(OP)                                   \
+  template<typename U>                                                \
+  constexpr vect<T, Size> &operator OP##=(vect<U, Size> const &other) \
+  {                                                                   \
+    for (std::size_t i = 0u; i < Size; ++i)                           \
+      array[i] OP## = other[i];                                       \
+    return (*this);                                                   \
+  }                                                                   \
+                                                                      \
+  template<typename U>                                                \
+  constexpr auto operator OP(vect<U, Size> const &other) const        \
+  {                                                                   \
+    vect<decltype(array[0] OP other[0]), Size> result{*this};         \
+                                                                      \
+    result OP## = other;                                              \
+    return result;                                                    \
+  }                                                                   \
+                                                                      \
+  template<typename U>                                                \
+  constexpr vect<T, Size> &operator OP##=(U const &other)             \
+  {                                                                   \
+    for (auto &elem : *this)                                          \
+      elem OP## = other;                                              \
+    return *this;                                                     \
+  }                                                                   \
+                                                                      \
+  template<typename U>                                                \
+  constexpr auto operator OP(U const &other) const                    \
+  {                                                                   \
+    vect<decltype(array[0] OP other), Size> result{*this};            \
+                                                                      \
+    result OP## = other;                                              \
+    return result;                                                    \
   }
 
     CLAWS_VECT_OPERATOR_DEF(+);
+
     CLAWS_VECT_OPERATOR_DEF(-);
+
     CLAWS_VECT_OPERATOR_DEF(*);
+
     CLAWS_VECT_OPERATOR_DEF(/);
+
     CLAWS_VECT_OPERATOR_DEF(%);
-    CLAWS_VECT_OPERATOR_DEF(^);
+
+    CLAWS_VECT_OPERATOR_DEF (^);
+
     CLAWS_VECT_OPERATOR_DEF(|);
+
     CLAWS_VECT_OPERATOR_DEF(&);
 
 #undef CLAWS_VECT_OPERATOR_DEF
 
-    // TODO optimize ?
-    constexpr bool equals(vect<dim, T> const &other) const
+    template<typename U>
+    constexpr bool operator==(vect<U, Size> const &other) const noexcept
```

Comment:
I'd prefer == be component-wise too.

Reply:
I'd rather keep operator overloads to their expected signature.
I'll add a `is_equal` which will be component-wise.

Reply:
It seems counter intuitive to have == not behave like >= <= > and <

Reply:
That is why there are no >= <= > and <, see the "comparators" test case

Reply:
Ok, sold. I believe the writing overhead is small, and even if I'm not convinced by this particular situation, the fact that their standard usage is sorting is enough to justify it for me.